packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -493,6 +496,24 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
           },
       );
 
+      registerServiceExtension(name: 'invokePreHotRestartCallbacks', callback: (Map<String, Object> params) async {
+        Future<void> invokeAndWait(PreHotRestartCallback callback, String? label) async {
+          try {
+            await callback();
+          } catch (error) {
+            debugPrint('Failed to invoke preHotRestartCallback ${label ?? 'unlabeled'}: $error');
```

Comment:
we have logic for this (FlutterError and so forth). If for some reason we can't use it here it would be good to leave a comment saying why.

Reply:
done